### PR TITLE
Feature: download results in Excel (XSLX) format (closes #801)

### LINKED ISSUE #801: Download DataSheets as Excel file.
Csv file with utf-8 is hard to use in excel. So I want to download data sheets as .xlsx file

### PR DESCRIPTION
**Pull request by ninneko**
Add download excel button at query result view.

### CODE REVIEW

File: redash/handlers/query_results.py [hunk 1]
```diff
@@ -137,6 +139,32 @@ def make_csv_response(query_result):
...
+    @staticmethod
+    def make_excel_response(query_result):
+        s = cStringIO.StringIO()
+
+        query_data = json.loads(query_result.data)
+        book = xlsxwriter.Workbook(s)
+        sheet = book.add_worksheet("result")
+        logging.info(query_data['columns'])
+
+        list = []
```
Comment (arikfr) [inline redash/handlers/query_results.py]: `list` is a builtin function in Python. Better to name it with a different name, maybe `columns_list`?

File: redash/handlers/query_results.py [hunk 2]
```diff
@@ -137,6 +139,32 @@ def make_csv_response(query_result):
...
         return make_response(s.getvalue(), 200, headers)
 
+    @staticmethod
+    def make_excel_response(query_result):
+        s = cStringIO.StringIO()
+
+        query_data = json.loads(query_result.data)
+        book = xlsxwriter.Workbook(s)
+        sheet = book.add_worksheet("result")
+        logging.info(query_data['columns'])
```
Comment (arikfr) [inline redash/handlers/query_results.py]: Many production deployments run with `INFO` level logging, and all these logging messages will significantly spam the logs. Better to switch to `DEBUG` or even remove them at all, as I doubt they will have value beyond development.

(this refers to all the info level messages in this method)

File: rd_ui/app/views/query.html [hunk 3]
```diff
@@ -158,11 +158,18 @@
             <hr>
 
             <p>
-                <a class="btn btn-primary btn-sm" ng-disabled="queryExecuting || !queryResult.getData()" query-result-link target="_self">
+            <p>
+                <a class="btn btn-primary btn-sm" ng-disabled="queryExecuting || !queryResult.getData()" query-result-csv-link target="_self">
                     <span class="glyphicon glyphicon-cloud-download"></span>
-                    <span>Download Dataset</span>
+                    <span>Download Csv</span>
```
Comment (arikfr) [inline rd_ui/app/views/query.html]: Csv -> CSV...

File: rd_ui/app/scripts/directives/query_directives.js [hunk 4]
```diff
@@ -58,6 +58,26 @@
     }
   }
 
+  function queryResultExcelLink() {
```
Comment (arikfr) [inline rd_ui/app/scripts/directives/query_directives.js]: We can use the same method for both CSV and XSLX links, by passing the file type as an attribute. But if you are not familiar with Angular enough, leave as is and I will refactor it myself afterwards.

**Comment by arikfr** [comment]
@ninneko thanks! This is a great addition.

**Comment by toyama-h** [comment]
Awesome :blush:

**Comment by ninneko** [comment]
Thank you for reviews.
I've fix it and the test case.

**Comment by arikfr** [comment]
Merged. Thank you for this great feature! (and for fixing the bug in tests)

**Pull request merged**